proto/src/domaintype.rs

```diff
@@ -33,6 +33,13 @@
 //! Note that the Prost library and the TryFrom method have their own set of errors. These are
 //! merged into a custom Error type defined in this crate for easier handling.
 //!
+//! Requirements:
+//! * The DomainType trait requires the struct to implement the Clone trait.
+//! * The DomainType trait requires the struct to have a #[rawtype(MYRAWTYPE)] attribute set where
+//!   the MYRAWTYPE is a structure that has the prost::Message trait implemented. (protobuf type)
+//! * The DomainType trait requires that the TryFrom<RawType> implemented on the structure has an
+//!   error type that implements Into<BoxError>. (The current implementations with anomaly are
+//!   fine.)
```

Comment:
Actually some of the comments need updating now.

Reply:
I'll do that.

Reply:
Minor updates in the documentation: removed references to the DomainType macro. I hope this is it. :)